### ISSUE TITLE
Unable to train pose detection model with coco json

### ISSUE THREAD

**Issue by Zumbalamambo** [ultralytics/ultralytics#12964]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

WARNING ⚠️ No labels found in /home/hardy/Documents/workspace/pose_dataset/uncompressed/labels/val2017.cache, training may not work correctly. 

### Environment

CUDA: 11.8
OS: Ubuntu 20.0.4
Ultralytics v: latest

### Additional

I'm looking forward to training the pose model with coco whole body dataset. 

1. I have downloaded `train2017.zip`, `val2017.zip` from [here](https://cocodataset.org/#download). I have downloaded `coco_wholebody_train_v1.0.json` and `coco_wholebody_val_v1.0.json` as well.
2. I converted the coco wholebody json to yolo format with this code

```
from ultralytics.data.converter import convert_coco

convert_coco(labels_dir="/home/user/dataset/labels", use_keypoints=True, use_segments=False,cls91to80=False)
```

3. I tried to train using this code
```
from ultralytics import YOLO

# Load a model
model = YOLO("yolov8s-pose.yaml")  # build a new model from YAML

# Train the model
results = model.train(data="dataset.yaml",
                      model="yolov8s-pose.yaml", epochs=100, imgsz=640, task='pose')
```

However, the Precision, recall is also 0 all the time even after running for about 30 epochs. Also it's ignoring all the labels that has been converted calling that those labels are corrupt!. How do I fix it? what am I doing wrong?

Here is the dataset.yaml for your reference,
```
# Ultralytics YOLO 🚀, AGPL-3.0 license
# COCO8-pose dataset (first 8 images from COCO train2017) by Ultralytics
# Documentation: https://docs.ultralytics.com/datasets/pose/coco8-pose/
# Example usage: yolo train data=coco8-pose.yaml
# parent
# ├── ultralytics
# └── datasets
#     └── coco8-pose  ← downloads here (1 MB)

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: /home/user/dataset/ # dataset root dir
train: images/train2017 # train images (relative to 'path') 4 images
val: images/val2017 # val images (relative to 'path') 4 images
test: # test images (optional)

# Keypoints
kpt_shape: [24, 3]  # number of keypoints, number of dims (2 for x,y or 3 for x,y,visible)
flip_idx: [0, 2, 1, 4, 3, 6, 5, 8, 7, 10, 9, 12, 11, 14, 13, 16, 15, 17, 19, 18, 20, 21, 23, 22]

# Classes
names:
  0: person
```

and here is the yolov8s.yaml

```
# Ultralytics YOLO 🚀, AGPL-3.0 license
# YOLOv8-pose keypoints/pose estimation model. For Usage examples see https://docs.ultralytics.com/tasks/pose

# Parameters
nc: 1  # number of classes
kpt_shape: [24, 3]  # number of keypoints, number of dims (2 for x,y or 3 for x,y,visible)
scales: # model compound scaling constants, i.e. 'model=yolov8n-pose.yaml' will call yolov8-pose.yaml with scale 'n'
  # [depth, width, max_channels]
  n: [0.33, 0.25, 1024]
  s: [0.33, 0.50, 1024]
  m: [0.67, 0.75, 768]
  l: [1.00, 1.00, 512]
  x: [1.00, 1.25, 512]

# YOLOv8.0n backbone
backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]]  # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]]  # 1-P2/4
  - [-1, 3, C2f, [128, True]]
  - [-1, 1, Conv, [256, 3, 2]]  # 3-P3/8
  - [-1, 6, C2f, [256, True]]
  - [-1, 1, Conv, [512, 3, 2]]  # 5-P4/16
  - [-1, 6, C2f, [512, True]]
  - [-1, 1, Conv, [1024, 3, 2]]  # 7-P5/32
  - [-1, 3, C2f, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]]  # 9

# YOLOv8.0n head
head:
  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 6], 1, Concat, [1]]  # cat backbone P4
  - [-1, 3, C2f, [512]]  # 12

  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 4], 1, Concat, [1]]  # cat backbone P3
  - [-1, 3, C2f, [256]]  # 15 (P3/8-small)

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 12], 1, Concat, [1]]  # cat head P4
  - [-1, 3, C2f, [512]]  # 18 (P4/16-medium)

  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 9], 1, Concat, [1]]  # cat head P5
  - [-1, 3, C2f, [1024]]  # 21 (P5/32-large)

  - [[15, 18, 21], 1, Pose, [nc, kpt_shape]]  # Pose(P3, P4, P5)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Zumbalamambo** [comment]
Also kindly please share if there are any models that has been trained with coco-wholebody dataset that can detect hand and feet similar to openpose

**Comment by glenn-jocher** [comment]
Hello! Currently, Ultralytics does not provide pre-trained models specifically trained on the COCO-WholeBody dataset for hand and feet detection similar to OpenPose. However, you can train a custom model using the COCO-WholeBody dataset by converting the dataset to the required YOLO format and adjusting the keypoints for your specific needs, such as focusing on hands and feet.

Here's a brief example of how you might start training a model for this purpose:

```python
from ultralytics import YOLO

# Load a pose estimation model configuration
model = YOLO("yolov8n-pose.yaml")

# Train the model on your dataset
results = model.train(data="your_dataset.yaml", epochs=100, imgsz=640)
```

Make sure your dataset YAML file is correctly set up to include the keypoints for hands and feet. If you need further customization or assistance, feel free to ask! 😊

**Comment by Zumbalamambo** [comment]
@glenn-jocher thank you! The yaml files are all correct. I get this error on all labels,

`val: WARNING ⚠️ /home/user/dataset/images/val2017/000000581317.jpg: ignoring corrupt image/label: labels require 77 columns each
`

**Comment by Zumbalamambo** [comment]
@glenn-jocher i also tried using `[JSON2YOLO](https://github.com/ultralytics/JSON2YOLO)` but I'm getting the same error. I've verified the paths of the yaml file, images, labels and they are in place.

**Comment by glenn-jocher** [comment]
@Zumbalamambo hi there! It seems like there might be a mismatch in the expected format for the labels. The error message indicates that each label should have 77 columns, which suggests your keypoints might not be formatted correctly or you're missing some data points. Could you double-check the format of your converted labels? They should match the expected structure, including all keypoints and visibility flags if used.

Here's a quick reminder of the format:
```
<class-index> <x_center> <y_center> <width> <height> <kpt1_x> <kpt1_y> <visibility1> ... <kpt24_x> <kpt24_y> <visibility24>
```
Make sure each line in your label files has the correct number of entries. If you continue to face issues, could you share a snippet of your label file for further inspection? 😊

**Comment by Zumbalamambo** [comment]
thank you! 

here is the [link](https://drive.google.com/drive/folders/1wBrYe83yo9JZpIx69J6798mOp0zBcIwz?usp=drive_link) to the val json file and the converted label

I used the official code to convert,
```

from ultralytics.data.converter import convert_coco

convert_coco(labels_dir="./data/labels", use_keypoints=True, use_segments=False,cls91to80=True)
```

labels folder has got the train and val jsons.

**Comment by glenn-jocher** [comment]
Hello! Thanks for sharing the link and details. It looks like you're on the right track with the conversion. If you're still encountering issues with the label format, please ensure that the `labels_dir` path in your `convert_coco` function correctly points to the directory containing your JSON files. Also, double-check that the `use_keypoints` parameter aligns with the data in your JSON files, particularly that keypoints data is indeed present.

If the problem persists, could you please check a few lines from your converted label files to ensure they match the expected format? Sometimes, a quick look at the actual data can help pinpoint any discrepancies. 😊
